sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -964,17 +964,14 @@ class SubquerySuite extends QueryTest
             |               WHERE t1.c1 = t2.c1)
           """.stripMargin)
       }
-      checkErrorMatchPVals(
+      checkError(
         exception1,
-        errorClass = "UNSUPPORTED_SUBQUERY_EXPRESSION_CATEGORY.UNSUPPORTED_CORRELATED_REFERENCE",
-        parameters = Map("treeNode" -> "(?s).*"),
-        sqlState = None,
+        errorClass = "UNSUPPORTED_SUBQUERY_EXPRESSION_CATEGORY.CORRELATED_REFERENCE",
+        parameters = Map("sqlExprs" -> "outer(arr_c2#427264)"),
```

Comment:
the same question here. Can we guarantee the same number across multiple runs.

Reply:
Thanks for catching out, seems like we should use regex here. Let me address